soda_machine.rb

```diff
@@ -7,9 +7,17 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
```

Comment:
Since there is an attr_reader for @sodas, you could drop the @ here and in subsequent instance methods